ledger-core-cosmos/src/cosmos/transaction_builders/CosmosLikeTransactionBuilder.cpp

```diff
@@ -366,10 +341,24 @@ namespace ledger {
                 if (feeObject[kAmount].IsArray()) {
                     auto fee = BigInt();
 
+                    auto getAmount = [=] (const Object &object) -> Amount {
+                        auto denom = getString(object, kDenom);
+                        auto amount = getString(object, kAmount);
+
+                        auto unit = std::find_if(currency.units.begin(), currency.units.end(), [&] (const api::CurrencyUnit &unit) {
+                            return unit.name == denom;
+                        });
+                        if (unit == currency.units.end()) {
+                            throw Exception(api::ErrorCode::INVALID_ARGUMENT, "Unknown unit while parsing transaction");
+                        }
+                        //TODO: Fix Amount::toUnit
+                        return Amount(currency, 0, BigInt(amount) * BigInt(10).powu(static_cast<unsigned short>((*unit).numberOfDecimal)));
+                    };
+
                     // accumlate all types of fee
                     for (auto& amount : feeObject[kAmount].GetArray()) {
                         if (amount.IsObject()) {
-                            fee = fee +  BigInt(getAmount(amount.GetObject()).toString());
+                            fee = fee + BigInt(getAmount(amount.GetObject()).toString());
```

Comment:
We will need a unit check if we add fees like that

Reply:
Done with an assert in getAmount() method.